tests/unit_tests/vecpar/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2022 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+
+# make unit tests for multiple algebras
+set( algebras "array" )
+if( DETRAY_EIGEN_PLUGIN )
+    list( APPEND algebras "eigen" )
+endif()
+
+### CPU tests
+find_package(OpenMP QUIET)
+if ((TARGET vecpar::all) AND (TARGET vecpar::omp))
+    foreach(algebra ${algebras})
+       # Unit tests for the selected algebra.
+       detray_add_test( ${algebra}_vecpar_cpu
+               "rk_stepper_vecpar.cpp" "rk_stepper_vecpar.hpp"
+               LINK_LIBRARIES GTest::gtest_main vecmem::cuda detray_tests_common
+               detray::${algebra} vecpar::all OpenMP::OpenMP_CXX)
+
+       target_compile_definitions( detray_test_${algebra}_vecpar_cpu
+               PRIVATE ${algebra}=${algebra} )
+
+       target_compile_options( detray_test_${algebra}_vecpar_cpu PUBLIC
+               $<$<COMPILE_LANGUAGE:CXX>:-std=c++2a>)
```

Comment:
Nope. Let's not do it like this.

If you want to compile this source file using C\+\+20, then use:

```cmake
set_target_properties( detray_test_${algebra}_vecpar_cpu
   PROPERTIES CXX_STANDARD 20 )
```

This way CMake will be able to choose the compiler-appropriate flag(s) for making this happen.

Reply:
You are right, this can be done better. I corrected it.